posts/inside-rust/2024-12-09-test-infra-nov-2024.md

```diff
@@ -0,0 +1,134 @@
+---
+layout: post
+title: "This Month in Our Test Infra: November 2024"
+author: Jieyou Xu
+team: the Bootstrap Team <https://www.rust-lang.org/governance/teams/infra#team-bootstrap>
+---
+
+# This Month in Our Test Infra: November 2024
+
+<!-- time period: 2024-11-05 through 2024-12-03 -->
+
+This is a quick summary of the changes in the test infrastructure for the
+[rust-lang/rust][r-l/r] repository[^scope] for **November 2024**. It also
+includes brief descriptions of on-going work.
+
+[^scope]: The test infra here refers to the test harness [compiletest] and
+supporting components in our build system [bootstrap]. This test infra is used
+mainly by rustc and rustdoc. Other tools like cargo, miri or rustfmt maintain
+their own test infra.
+
+As usual, if you encounter bugs or UX issues when using our test infrastructure,
+please [file an issue][new-issue]. Bugs and papercuts can't be fixed if we don't
+know about them!
+
+**Thanks to everyone who contributed to our test infra!**
+
+## Highlights
+
+### compiletest: Add `proc-macro` auxiliary build directive
+
+[@ehuss](https://github.com/ehuss) added a `//@ proc-macro` directive that
+behaves like `//@ aux-build`, but it packages the usual `//@ force-host` and
+`//@ no-prefer-dynamic` boilerplate that previously was needed by proc-macro
+auxiliaries. If the main test file also uses a sufficiently new edition (i.e.
+Edition 2018 onwards), the proc-macro auxiliary is also made available via
+extern prelude.
+
+**Before**: test writer need to write `//@ force-host` and `//@
+no-prefer-dynamic` for each and every proc-macro auxiliary.
+
+```rs
+// tests/ui/foo/my-main-test.rs
+//@ aux-build: my-proc-macro.rs
+```
+
+```rs
+// tests/ui/foo/auxiliary/my-proc-macro.rs
+//@ no-prefer-dynamic
+//@ force-host
+```
+
+**After**: only `//@ proc-macro` directive is needed in main test file.
+
+```rs
+// tests/ui/foo/my-main-test.rs
+//@ proc-macro: my-proc-macro.rs
+```
+
+```rs
+// tests/ui/foo/auxiliary/my-proc-macro.rs
+```
+
+Thanks Eric!
+
+### rustc: make `rustc` consider itself a stable compiler when `RUSTC_BOOTSTRAP=-1` is set
+
+In [#132993](https://github.com/rust-lang/rust/pull/132993), I modified
```

Comment:
Just a drive-by: Maybe using @jieyouxu instead of `I`?

* Make the post in a more neutral tone.
* Can immediately know who worked on this when reading

Reply:
I mean the post author is right there so lol

Reply:
I feel like calling yourself out gives the right amount of credits to your work. People just likely pay less attention to post authors 😞

It's up to you anyway. I love the post. Thank you :)